script/rubocop-autocorrect.sh

```diff
@@ -25,18 +36,38 @@ git commit --all --message "Regenerate Rubocop's TODO file"
 # mode to find the next failing cop.
 grep "This cop supports safe autocorrection" -A 5 .rubocop_todo.yml\
   | grep '^[A-Z]'\
-  | head "${@:1}"\
   | tr -d :\
-  | while read cop; do
+  | while read cop && [ "$files_per_branch" -gt 0 ]; do
       echo "Trying to autocorrect safely: $cop"
       bundle exec rubocop --regenerate-todo --except "$cop"
 
       echo "Safely autocorrect $cop" > .git/COMMIT_EDITMSG
       echo "" >> .git/COMMIT_EDITMSG
       bundle exec rubocop --autocorrect >> .git/COMMIT_EDITMSG
       if grep -q "offenses corrected" .git/COMMIT_EDITMSG; then
-        git add --all
+
+        unstaged_files=$(git status -s -uno | wc -l)
+        remaining_files=$((unstaged_files - files_per_branch))
+
+        # <dev/tty is needed since stdin is pointing at the file
+        if ((unstaged_files > files_per_branch)) && get_user_confirmation $remaining_files $cop </dev/tty; then
+          git add .
+        else
+          # Commit up to the limit of files defined in the parameter
+          echo "hwewlwkj"
+          git status --porcelain | grep '^ M' | head -n $files_per_branch | awk '{print $2}' | xargs git add
+        fi
+
+        changed_files=$(git diff --name-only --cached | sed '/^$/d' | wc -l)
+        files_per_branch=$((files_per_branch - changed_files))
+
         git commit --file .git/COMMIT_EDITMSG
+
+        # Discard unstage files once we hit the limit
+        if ((files_per_branch <= 0)); then
+          git restore .
```

Comment:
My version of Git doesn't have this command.

Reply:
It was introduced in 2.27 (June 2020). There is a more round about way to do it of course, but this a handy one-liner.